Extensions/XMPPMUCSUB/XMPPMUCSUB.m

```diff
@@ -750,36 +751,36 @@ - (void)xmppStream:(XMPPStream *)sender didReceiveMessage:(XMPPMessage *)message
      */
     
     NSXMLElement *event = [message elementForName:kXMPPEvent xmlns:XMPPPubSubEventNamespace];
-    NSXMLElement *messageItems = [event elementForName:kXMPPItems xmlns:XMPPMucSubMessageNamespace];
-    NSXMLElement *presenceItems = [event elementForName:kXMPPItems xmlns:XMPPMucSubPresenceNamespace];
-    NSXMLElement *subscribeItems = [event elementForName:kXMPPItems xmlns:XMPPMucSubSubscribeNamespace];
-    NSXMLElement *unsubscribeItems = [event elementForName:kXMPPItems xmlns:XMPPMucSubUnsubscribeNamespace];
-    
-    if (messageItems) {
-        NSXMLElement *item = [messageItems elementForName:kXMPPItem];
-        NSXMLElement *messageElement = [item elementForName:kXMPPMessage];
-        if (messageElement) {
-            XMPPMessage *mucSubMessage = [XMPPMessage messageFromElement:messageElement];
-            [multicastDelegate xmppMUCSUB:self didReceiveMessage:mucSubMessage];
+    NSXMLElement *items = [event elementForName:kXMPPItems];
+    
+    if (items) {
+        NSString *node = [[items attributeForName:kXMPPNode] stringValue];
+        if ([node isEqualToString:XMPPMucSubMessageNamespace]) {
+            NSXMLElement *item = [items elementForName:kXMPPItem];
```

Comment:
You can pull `item` out of the if/elseif/elseif block

Reply:
√